multisrc/src/main/java/eu/kanade/tachiyomi/multisrc/wpcomics/WPComics.kt

```diff
@@ -94,7 +97,7 @@ abstract class WPComics(
                 addQueryParameter("sort", "0")
             }
 
-            GET(url.toString().replace("/tim-truyen?status=2&", "/truyen-full?"), headers)
+            GET(url.toString().replace(replaceSearchPathOld, replaceSearchPathNew), headers)
```

Comment:
I'm not sure this is actually accomplishing something. NhatTruyen overrides searchPath; so the `replace()` as it was should only be effecting NetTruyen.

Reply:
yes, currently it only affects NetTruyen but what if another source added in future or in another fork decides to use the same "tim-truyen" as `searchPath`? They would have to overwrite the whole `searchMangaRequest` function with same code as the generator just to avoid the replacement.
Or like XoxoComics has to overwrite the `searchMangaRequest` function instead of only need to overwrite `searchPath`.

Because the replacement code was there previously so I only fix it and keep its position. But actually maybe it should be placed inside NetTruyen only since only it needs that. NetTruyen would still need to overwrite the whole `searchMangaRequest` function though.

How do you think it should be?

Reply:
Originally WPComics came together as a hodgepodge of a few sources and `searchMangaRequest()` was a basic implementation. It was slowly reworked to add filters and accommodate the sources that relied on it; no one got around to refactoring so that XoxoComics wouldn't need to override. In other words, it ended up that way rather than it needed to be that way.

I kind of think it'd be better to wait for an actual case where the `replace()` as it is creates a conflict rather than changing for a hypothetical case. What if new sources get added that would have worked with the `replace()` as it was? Then those new sources have to include overrides for your new `vals` when they could have worked without having to add overrides.

Reply:
It's not really a hypothetical case for me. My fork has 4 different sources: nettruyenss.com, nhattruyento.com, nettruyenx.com & nhattruyens.com. Only nettruyenss.com needs to use the replace. That means the replace should not be default for the generator but should be used only when a source needs it.

I think one should take as less effort as possible to add a new source. Leaving the replace there, to me, is kind of like a trap for future contributors/maintainers :) Just trying to save some efforts